test/metabase/cmd/compare_h2_dbs.clj

```diff
@@ -112,7 +119,19 @@
 (defn different-contents?
   "Diff contents of 2 DBs. Returns truthy if there is a difference, falsey if not."
   [db-file-1 db-file-2]
-  (jdbc/with-db-connection [conn-1 (jdbc-spec db-file-1)]
-    (jdbc/with-db-connection [conn-2 (jdbc-spec db-file-2)]
-      (or (different-table-names? conn-1 conn-2)
-          (different-rows? conn-1 conn-2)))))
+  (let [spec-1 (jdbc-spec db-file-1)
+        spec-2 (jdbc-spec db-file-2)]
+    (sql-jdbc.execute/do-with-connection-with-options
+     driver/*driver*
+     spec-1
+     nil
+     (fn [db-1-connection]
```

Comment:
A mad a slight tweak to this, you were using `do-with-connection-with-options` but ignoring the connection returned by it, which meant the it probably would have been opening new connections again in the calls to `different-table-names?` and `different-rows?`. Tweaked so it uses the existing open connections.

Reply:
Thanks, it makes sense how you use the existing connections in https://github.com/metabase/metabase/pull/34268/commits/3cb72963b774761753a4afa83d1b621a61ad34f0